### PR TITLE
Convert go plugin example to GPG based

### PR DESCRIPTION
**Pull request by yujunz**

### CODE REVIEW

**Comment by Liujingfang1** [comment]
Can we keep the existing one and adding the `gpg` based secret generator? Maybe in the same doc or a separate doc. I'm fine with both.

**Comment by yujunz** [comment]
> Can we keep the existing one and adding the `gpg` based secret generator? Maybe in the same doc

Done

File: docs/plugins/goPluginGuidedExample.md [hunk 1]
```diff
@@ -232,31 +237,46 @@ generators:
 EOF
 ```
 
-Now for the hard part.  Generate the real encrypted data.
+Now generate the real encrypted data.
 
+### Assure you have an encrytion tools installed
```
Comment (Liujingfang1) [inline docs/plugins/goPluginGuidedExample.md]: tools --> tool

Comment (Liujingfang1) [inline docs/plugins/goPluginGuidedExample.md]: encrytion --> encryption

**Pull request merged**